src/core/lib/security/credentials/external/external_account_credentials.cc

```diff
@@ -197,6 +199,30 @@ RefCountedPtr<ExternalAccountCredentials> ExternalAccountCredentials::Create(
       return nullptr;
     }
   }
+  int32_t lifetime = IMPERSONATED_CRED_DEFAULT_LIFETIME_IN_SECONDS;
+  it = json.object().find("service_account_impersonation");
+  if (it != json.object().end()) {
+    auto service_acc_imp_obj_it =
+        it->second.object().find("token_lifetime_seconds");
+    if (service_acc_imp_obj_it != it->second.object().end()) {
+      if (service_acc_imp_obj_it->second.type() != Json::Type::kNumber) {
+        *error = GRPC_ERROR_CREATE("token_lifetime_seconds must be a number.");
+        return nullptr;
+      }
+      lifetime =
+          strtol(service_acc_imp_obj_it->second.string().c_str(), nullptr, 10);
+      if (lifetime > TWELVE_HOURS_IN_SECONDS) {
+        *error = GRPC_ERROR_CREATE(
+            "token_lifetime_seconds must be less than 43200s.");
+        return nullptr;
+      }
+    }
+  }
+  Json::Object service_account_impersonation;
```

Comment:
Can we not just do
options.service_account_impersonation = json.object().find("service_account_impersonation") instead of reading in the json and recreating it? And then use the default in the call to iamcredentials endpoint instead of inserting it here.